### ISSUE TITLE
License - AWS vs. Apache 2

### ISSUE THREAD

**Issue by etspaceman** [WW-Digital/reactive-kinesis#57]
See this thread:

https://github.com/akka/alpakka/pull/434

Alpakka determined that they were unable to continue to distribute their software under the Apache 2 license when leveraging the KCL library. This is because the KCL is distributed with its own License.

This repository is facing the same conundrum. I think that the license needs to be changed to match the AWS Licnese for it to be legally compliant w/ their SDK distribution:

https://github.com/awslabs/amazon-kinesis-client/blob/master/LICENSE.txt

CC @markglh

**Comment by markglh** [comment]
Thanks!

We `might` be ok here - reading around at what others have done. I think Alpakka is an interesting one because it includes a lot of other connectors. But we're solely focussed on Kinesis. I'm not expert in this field though.

This is how I understood it:
https://issues.apache.org/jira/browse/LEGAL-198?focusedCommentId=15618639&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-15618639

Maybe we should do something like this?
https://github.com/localstack/localstack#license

**Comment by markglh** [comment]
I've had a discussion about this internally (Bamtech/Cake, not WW) - we may have to go with this... But first I'm gonna bitch and moan at amazon via internal channels - watch this space.

/cc @agaro1121 @Nick-Smith

**Comment by etspaceman** [comment]
Yeah I'm not happy about it either. Replacing an Apache 2 license is never fun 😟

**Comment by etspaceman** [comment]
Just curious - any movement on whether or not this is needed?

**Comment by markglh** [comment]
I think it is but I never heard back officially. I'll chase again!

**Comment by etspaceman** [comment]
The KCL has recently upgraded their license to Apache 2.0:

https://github.com/awslabs/amazon-kinesis-client/releases/tag/v1.10.0
